test/module-test.cc

```diff
@@ -55,11 +76,551 @@ TEST(module_test, detail_namespace) {
 // macros must not be imported from a *named* module  [cpp.import]/5.1
 TEST(module_test, macros) {
 #if defined(FMT_HIDE_MODULE_BUGS) && \
-  defined(_MSC_FULL_VER) && _MSC_FULL_VER <= 192930036
-// bug in msvc 16.10-pre4:
+  defined(_MSC_FULL_VER) && _MSC_FULL_VER <= 192930129
+// bug in msvc up to 16.11-pre1:
 // include-guard macros leak from BMI
 // and even worse: they cannot be #undef-ined
   macro_leaked = false;
 #endif
   EXPECT_FALSE(macro_leaked);
 }
+
+// The following is less about functional testing (that's done elsewhere)
+// but rather visibility of all client-facing overloads, reachability of
+// non-exported entities, name lookup and overload resolution within
+// template instantitions.
+// Excercise all exported entities of the API at least once.
+// Instantiate as many code paths as possible.
+
+TEST(module_test, to_string) {
+  EXPECT_EQ("42", fmt::to_string(42));
+  EXPECT_EQ("42", fmt::to_string(42.0));
+}
+
+TEST(module_test, format) {
+  EXPECT_EQ("42", fmt::format("{:}", 42));
+  EXPECT_EQ("-42", fmt::format("{0}", -42.0));
+}
+
+TEST(module_test, format_to) {
+  std::string s;
+  fmt::format_to(std::back_inserter(s), "{}", 42);
+  EXPECT_EQ("42", s);
+
+  char buffer[4] = {0};
+  fmt::format_to(buffer, "{}", 42);
+  EXPECT_EQ("42", std::string_view(buffer));
+
+  fmt::memory_buffer mb;
+  fmt::format_to(mb, "{}", 42);
+  EXPECT_EQ("42", std::string_view(buffer));
+}
+
+TEST(module_test, formatted_size) {
+  EXPECT_EQ(2u, fmt::formatted_size("{}", 42));
+}
+
+TEST(module_test, format_to_n) {
+  std::string s;
+  auto result = fmt::format_to_n(std::back_inserter(s), 1, "{}", 42);
+  EXPECT_EQ(2u, result.size);
+  char buffer[4] = {0};
+  fmt::format_to_n(buffer, 3, "{}", 12345);
+}
+
+TEST(module_test, format_args) {
+  auto no_args = fmt::format_args();
+  EXPECT_FALSE(no_args.get(1));
+
+  using ctx = fmt::format_context;
+  fmt::basic_format_args<ctx> args = fmt::make_format_args<ctx>(42);
```

Comment:
`ctx` can be dropped here because it is the default. Same in multiple places below.